enhancements/run-bundle.md

```diff
@@ -0,0 +1,263 @@
+packagemanifest. The change to `run packagemanifest` is covered by
+[OSDK-1126](https://issues.redhat.com/browse/OSDK-1126).
+
+## Proposal
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Operator SDK currently supports running an operator with OLM using the
+`run` command with the `--olm` flag. The existing command only supports the
+package manifest format. This enhancement will add a new subcommand, `bundle`,
+to the `run` command. This new command, `run bundle`, will support the new
+bundle format. More information about the different formats can be found at
+the URLs below:
+
+ * Operator bundles: https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+ * Package manifests format: https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+
+The idea is to consolidate the run command into different subcommands to make
+things easier to understand and eliminate flag bloat.
+
+ * `run local` - exists today
+ * `run packagemanifest` - covered by [OSDK-1126](https://issues.redhat.com/browse/OSDK-1126)
+ * `run bundle` - covered by this design document
+
+#### Command Line Interface
+The new command line will look as follows:
+
+```
+operator-sdk run bundle <bundle-image> [--index-image=] [--namespace=] [--install-mode=(AllNamespace|OwnNamespace|SingleNamespace=)]
+```
+
+* `<bundle-image>` is a positional argument that specifies the bundle image.
+* `[--index-image=]` is an optional flag that specifies the index image.
+  * If `--index-image` is not provided, default to
+    `quay.io/operator-framework/upstream-registry-builder`
+  * If `--index-image` is provided, use the provided value
+* `[--namespace=]` is an optional flag that specifies the namespace to install
+  the operator. It will default to the `KUBECONFIG` context.
+* `[--install-mode=]` is an optional flag that specifies the
+  [OLM install mode](https://red.ht/2YK0ksD).
+  * We support 3 modes:
+    * AllNamespaces
+    * OwnNamspace
+    * SingleNamespace=ns1
+      * NOTE: if ns1 is the same as the operator's namespace, a warning will be
+        printed
+  * When `--install-mode` is not supplied, the default precendence order is:
+    * `OperatorGroup`, if it already exists
+    * `AllNamespaces`, if the CSV supports it
+    * `OwnNamespace`, if the CSV supports it
+    * If the CSV does not support either of these, `--install-mode` flag
+      will be **required**
+
+#### Assumptions
+Like the existing `run` command, we **assume** the following:
+
+ 1. OLM is already installed
+ 1. kube API is available with the permissions to create resources
+ 1. kube server can see and pull the provided images (i.e. bundle and index)
+ 1. If an index image is provided, we assume that it has:
+    * bash, so we can change the entrypoint
+    * dependencies required for `opm registry add` so we can inject
+       the bundle into the DB
+    * the existing registry DB located at `/database/index.db`
```

Comment:
Is there a technical reason you can't infer the database path on the image? We have a spec that defines a path label for the index image.

Reply:
There is not. It would require a bit more work in the SDK to use the containerd tooling. We think (correct us if we're wrong) that most users will have built their indexes with OPM tooling that defaults to this location, so this works for an MVP.

We plan to do DB path detection could be done in a follow-up.

Reply:
That seems totally reasonable.

Reply:
Just to surface this again - if we plan to follow labels as a followup, shouldn't that be written here (for now assume static path, in the future use the label?)

Reply:
Updated non-goals to state the following:

For now we will assume the existing registry DB located at `/database/index.db`.
In a follow-up design we can investigate having the SDK use the labels to       
detect the DB path.